libraries/ms-common/src/stm32f0xx/adc.c

```diff
@@ -132,6 +132,29 @@ StatusCode adc_set_channel(ADCChannel adc_channel, bool new_state) {
   return STATUS_CODE_OK;
 }
 
+// ADC Channel to GPIO Address mapping found in table 13 of the specific device datasheet.
+// Channels 0 to 7 are occupied by port A, 8 to 9 by prt B, and 10 to 15 by port C
+StatusCode adc_get_channel(GPIOAddress address, ADCChannel *adc_channel) {
+  *adc_channel = address.pin;
+
+  switch (address.port) {
+    case GPIO_PORT_A:
+      if (address.pin > 7) { return STATUS_CODE_INVALID_ARGS; }
```

Comment:
We generally try not to have single-line expressions like this. Also, `status_code` please.